### ISSUE TITLE
CRC check of TLV/MMT stream

### ISSUE THREAD

**Issue by MMT-TLVassociation** [wangf1978/DumpTS#15]
https://github.com/MMT-TLVassociation/DumpTS/commit/c884656196b29790064bb09ad455a6f7470431a8
(In crc.cpp, const uint8_t* is changed to uint8_t* . )

I added code to check CRC in DumpTS.

example:
DumpTS file.mmts  --CID=1 --pid=0xF300 --output=e00301.hevc

If there are any CRC error in TLV/MMT stream, CRC false message is output.
I'll pull request if your are OK.

**Comment by wangf1978** [comment]
Hi, thanks for your contribution, it is ok for me, please go ahead.

**Comment by MMT-TLVassociation** [comment]
Thank you.

**Comment by wangf1978** [comment]
I am not familiar with github merge function from one fork to the master branch, can you help to push it to the master branch?

**Comment by wangf1978** [comment]
Your codebase seems not to be a fork of the original source depot, I merged your changes manually, and I found some typos, and I changed it, the main change is 
    ```
    if (peek_msg_id == 0x8000 || 0x8002 || 0x8003 || 0x8004)// Packet_id, which includes CRC32
    ```
Changed it to 
    ```
    if (peek_msg_id == 0x8000 || peek_msg_id == 0x8002 || peek_msg_id == 0x8003 || 0x8004)// Packet_id, which includes CRC32
    ```

**Comment by MMT-TLVassociation** [comment]
Yes, you are correct. Thank you for manually merging the changes.
